gspread/worksheet.py

```diff
@@ -927,7 +822,28 @@ def get(
             self.spreadsheet_id, range_name, params=params
         )
 
-        return ValueRange.from_json(response)
+        vals = response.get("values", [[]])
+
+        if pad_values is True:
+            try:
+                vals = fill_gaps(vals)
```

Comment:
We need to compute the requested size here then pass it to the `fill_gaps` function in order to extend the matrix `vals` otherwise the matrix `vals` is too small

Reply:
The padding behaviour is not currently in gspread. It will not be until https://github.com/burnash/gspread/pull/1305 is merged.

Thus, the code not padding is correct, and the same as it was before.

Reply:
should we first merge #1305  ? so then the integration here will be simpler

Reply:
We could, yes. We would then have to merge master into v6 branch, since this PR targets v6.